docs/rules/prefer-destructuring.md

```diff
@@ -33,7 +33,9 @@ This rule takes two sets of configuration objects; the first controls the types
 
 The first has two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently.  By default, both are `true`.
 
-The second has a single property, `enforceForRenamedProperties`, that controls whether or not the `object` destructuring rules are applied in cases where the variable requires the property being access to be renamed.
+The first has two properties, `VariableDeclarator` and `AssignmentExpression`, which can be used to turn on or off the destructuring requirement for each of those types independently. Each property accepts an object that accpets two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently for variable declarations and assignment expressions.  By default, `array` and `object` are set to true for both `VariableDeclarator` and `AssignmentExpression`.
+
+The second has a single property, `enforceForRenamedProperties`, that controls whether or not the `object` destructuring rules are applied in cases where the variable requires the property being access to be renamed. If `object` in `AssignmentExpression` is set to false, `enforceForRenamedProperties` property is ignored.
```

Comment:
Why is `enforceForRenamedProperties` ignored in this case? It seems like it would still be useful in variable declarations for something like

```js
var foo = bar.baz

// ->

var {baz: foo} = bar;
```